web-common/src/features/dashboards/time-dimension-details/charts/TDDAlternateChart.svelte

```diff
@@ -45,6 +51,20 @@
   $: comparedDimensionLabel =
     $comparisonDimension?.label || $comparisonDimension?.name;
 
+  $: hoveredTime = $tableInteractionStore.time;
+  $: hoveredDimensionValue = $tableInteractionStore.dimensionValue;
+
+  $: {
+    updateVegaOnTableHover(
+      viewVL,
+      chartType,
+      isTimeComparison,
+      hasDimensionData,
+      hoveredTime,
+      hoveredDimensionValue,
+    );
```

Comment:
**Performance Issue**: The function 'updateVegaOnTableHover' could be called excessively without proper throttling, leading to significant performance degradation, especially with complex Vega visualizations. <br> **Fix**: Implement throttling or conditional checks to minimize the frequency of updates, especially when 'hoveredTime' or 'hoveredDimensionValue' changes are minor. <br> **Code Suggestion**: 
 ```
 +  $: if (needToUpdateVega(hoveredTime, hoveredDimensionValue)) {
 +    updateVegaOnTableHover(viewVL, chartType, isTimeComparison, hasDimensionData, hoveredTime, hoveredDimensionValue);
 +  }
 ```